add gameflow docs (#25)
kana-master/kana-master: gameflow.md

# Gameflow

## Game Logic

> The selected level starts and presents the user a random syllable (Hiragna / Katakana) as well as four possible answers.

Reviewer: Typo: "Hiragna"

Reviewer: Also: by random you mean a randomly selected one within the level right? Maybe you could go even more specific here and mention at least an approximate number of syllables within a level, just to communicate the dimensions. The same goes for the segments down here as well.

Author: But in the line below it's mentioned that the available syllables depend on the selected level. Doesn't that explain it already?

The available syllables depend on the selected level.

If the correct answer was selected, the game immediately goes to the next syllable.

If a wrong answer was selected, the game shows a quick indicator that something was wrong, removes a heart container and then goes to the next syllable.

> If the required amount of syllables have been answered, the game navigates to the success screen and informs the player how many Gems / Jewels he won as well as if he unlocked new things like syllables.

Reviewer: Don't wanna be exclusive!
```suggestion
If the required amount of syllables have been answered, the game navigates to the success screen and informs the player how many Gems / Jewels they won as well as if they unlocked new things like syllables.
```

> If the user runs out of Heart Containers the game navigates to the failure screen.

Reviewer: ```suggestion
If the user runs out of heart containers the game navigates to the failure screen.
```

## Game Segments

**Syllable**

> A single piece of gameplay in which the player has to select the correct counterpart for a given syllable in latin character.

Reviewer: Do all answers have to be correct to pass on?

Author: See comment below.

**Level**

> A collection of syllables the user has to complete. At the end of the level the user will be awarded in the form of gems.

Reviewer: Note my point above concerning a rough estimate on how many Syllables we talk about here.

**Gems / Jewels**

Will be rewarded on completion of a level, no matter if the user succeeded or failed.

> Can be used to buy upgrades in the Gems / Jewels store. In this way the player can ease the growing complexitiy / difficulity of newly unlocked levels.

Reviewer: ```suggestion
Can be used to buy upgrades in the gems / jewels store. This way, the player can ease the growing complexity / difficulty of newly unlocked levels.
```

**Heart Container**

> Represents the amount of wrong answers a user is allowed to give per level. Will be refilled on start of a new level. Initially, 3 Heart Containers are given.

Reviewer: ```suggestion
Represents the amount of wrong answers a user is allowed to give per level. Will be refilled on start of a new level. Initially, three heart containers are given.
```

**Upgrades**

> - Additional Heart Containers

Reviewer: ```suggestion
- Additional heart containers
```

> - New and more valubale Gems / Jewels

Reviewer: ```suggestion
- New and more valuable gems / jewels
```

Reviewer: Do we perhaps want to add information here what this means? Like, what is the value of a Gem?
Also, shouldn't we settle for one name?

Author: Regarding value: We could, but this should be discussed first I think.
Regarding name: We should, but I figured this is worth discussing.

> - More time to give an answer

Reviewer: I would perhaps rephrase this a bit for clarity, something like:
```suggestion
- Increased maximum time per syllable question
```

## Level Object

A level object provides all the data needed to run the game logic.

```
{
  /*
   * Identifier of the current Level.
   * Represents the actual level the user selected.
   */
  id: 1,

  /*

>    * The amount of syllables the user has to answer

Reviewer: "has to answer" - _correctly_?

Author: Not `correctly`. My idea was that the user has to answer a specific amount of syllables, doesn't matter if correct or not. He just has to make it through all the required syllables without consuming all of the heart containers.
